### DISCUSSION TITLE
Freezes chromium process(?) on parallel startup with Gauge

### DISCUSSION THREAD

**Discussion by ghost** [getgauge/taiko#2632] [Development]
When run tests with gauge+taiko in parallel mod, random process (all platforms - mac, win, linux (docker)) freezes.

Start command: `gauge run -n=5 --parallel --only parallelStreams --max-retries-count=3`.

I don't understand where is the problem, but if open the task manager - I expect to see the number of chromium's processes equal from command (-n=5 or other). 
But i see only one chromium process. In console - test stops (need to cancel the command).

Last Gauge log in Docker:
<img width="1272" alt="Снимок экрана 2022-08-13 в 00 13 30" src="https://user-images.githubusercontent.com/66106419/184409804-754b0002-72e1-4439-b40d-b6bc33495766.png">

Last taiko log:
```
2022-08-12T18:08:51.9701511Z 2022-08-12T18:08:44.846Z taiko:event Response Recieved: Request id: 1541.1033
2022-08-12T18:08:51.9702175Z 2022-08-12T18:08:44.847Z taiko:event Request resolved:	 RequestId : 1541.1033
2022-08-12T18:08:51.9702979Z 2022-08-12T18:08:44.847Z taiko:event Response Recieved: Request id: 1541.996
2022-08-12T18:11:25.4942131Z ##[error]The operation was canceled.
```

I will be grateful for any help/ideas with debagging.

Open/Close browser code:
```
export default class BeforeSpecs {
  @BeforeSpec({tags: ['headless']})
  async openHeadlessBrowserBeforeSpec(context: ExecutionContext) {
    await openBrowserBeforeSpec(true, context);
  }

  @BeforeSpec({tags: ['headed']})
  async openHeadedBrowserBeforeSpec(context: ExecutionContext) {
    await openBrowserBeforeSpec(false, context);
  }
}

async function openBrowserBeforeSpec(
  isHeadless: boolean,
  context: ExecutionContext,
) {
  await openBrowser({
    headless: isHeadless,
    args: [
      '--disable-gpu',
      '--disable-dev-shm-usage',
      '--disable-setuid-sandbox',
      '--no-first-run',
      '--no-sandbox',
      '--no-zygote',
      '--start-fullscreen',
      '--window-size=1920,1080',
    ],
  });

  if (context.getCurrentSpec()?.getName() !== `Login`) {
    await goto(pageUrl.login);
    await login(
      testCredentials.correct.username,
      testCredentials.correct.password,
    );
    await assertAfterLoginPage();
  }
  await beforeSpecLog(context);
}
```

**Comment by NivedhaSenthil** [comment]
Can you please provide more insights on how is the code organised, specifically `openBrowser` and `closeBrowser` ? A sample will be really great to understand the scenario.

**Reply by ghost** [reply]
Sorry for the long answer. Added code example from BeforeSpecs hook for opening and closing the browser.
